forward/forwarder.go

```diff
@@ -50,6 +50,7 @@ type Options struct {
 	RerouteRetries bool
 	RetrySchedule  []time.Duration
 	Timeout        time.Duration
+	Headers        []byte
```

Comment:
@prashantv thanks for the response. Indeed, for HTTP, `arg2` has other uses too; and I think the same can be said for JSON since there is no restriction on having only string values in the object.

I feel good about the Ringpop forward API exposing headers as `[]bytes` but I understand how this can be confusing. @thanodnl suggested that I should add a few examples in Ringpop on how the headers forwarding works in practice. This will demo the idiomatic way of using this feature.

Alternatively, we also discussed adding an utility function for the headers encoding. But since each encoding is different (HTTP needs the method and the URL and JSON is not limited to `map[string]string`) I think this won't bring too much value to our users; educating them on how to forward `arg2` as bytes and what serialization is available already in tchannel is probably better.